src/sentry/api/endpoints/user_details.py

```diff
@@ -89,3 +97,69 @@ def put(self, request, user):
             return Response(serialize(user, request.user))
 
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    @sudo_required
+    def delete(self, request, user):
+        """
+        Delete User Account
+
+        Also removes organizations if they are an owner
+        :pparam string user_id: user id
+        :param list organizations: List of organization ids to remove
```

Comment:
i'm wondering if we should actually have people send a list or if we should just delete all of the ones that should be deleted automatically? do we allow people to delete their account without removing the single owner orgs?

Reply:
Yeah I was thinking the same thing - the existing behavior is that single owner orgs get listed and always get deleted.

If it is an org w/ multiple owners, you can choose to delete it (but maybe it's better to not be able to delete those at all?)

Reply:
oh interesting. i didn't realize that you could also delete non-single owner orgs from here. hmmmm i mean i think in the interest of plowing through the setting stuff, it's probably a bad idea to change behavior, but seems like it'd be better if we forced people through the delete org flow when it's not single owner since it's so destructive. soo i'll defer to you/maybe you could check with chris to see if he's against removing the ability to delete those orgs from here?

Reply:
I'd make it explicit (send the list of orgs to remove). Otherwise it should probably:

- delete orgs explicitly selected
- delete orgs with no owners
- retain orgs with owners and not selected

Org deletion at least queues for 24 hours and emails all owners a way to undo it.